src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1322,16 +1542,44 @@ private static unsafe nuint NarrowUtf16ToAscii_Sse2(char* pUtf16Buffer, byte* pA
             }
             else
             {
-                if ((Sse2.MoveMask(Sse2.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate).AsByte()) & NonAsciiDataSeenMask) != 0)
+                if (Sse2.IsSupported)
                 {
-                    return 0;
+                    if ((Sse2.MoveMask(Sse2.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate).AsByte()) & NonAsciiDataSeenMask) != 0)
+                    {
+                        return 0;
+                    }
+                }
+                else if (AdvSimd.Arm64.IsSupported)
+                {
+                    if (ContainsNonAsciiValue(AdvSimd.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate)))
+                    {
+                        return 0;
+                    }
+                }
+                else
+                {
+                    throw new PlatformNotSupportedException();
                 }
             }
 
             // Turn the 8 ASCII chars we just read into 8 ASCII bytes, then copy it to the destination.
 
-            Vector128<byte> asciiVector = Sse2.PackUnsignedSaturate(utf16VectorFirst, utf16VectorFirst);
-            Sse2.StoreScalar((ulong*)pAsciiBuffer, asciiVector.AsUInt64()); // ulong* calculated here is UNALIGNED
+            Vector128<byte> asciiVector;
+            if (Sse2.IsSupported)
+            {
+                asciiVector = Sse2.PackUnsignedSaturate(utf16VectorFirst, utf16VectorFirst);
+                Sse2.StoreScalar((ulong*)pAsciiBuffer, asciiVector.AsUInt64()); // ulong* calculated here is UNALIGNED
+            }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                Vector64<byte> lower = AdvSimd.ExtractNarrowingSaturateUnsignedLower(utf16VectorFirst);
+                asciiVector = AdvSimd.ExtractNarrowingSaturateUnsignedUpper(lower, utf16VectorFirst).AsByte();
```

Comment:
You don't need a call to `ExtractNarrowingSaturateUnsignedUpper` since you will store only lower 8 bytes to memory.

```c#
Vector64<byte> lower = AdvSimd.ExtractNarrowingSaturateUnsignedLower(utf16VectorFirst);
AdvSimd.Store(pAsciiBuffer, lower);
```